addons/mail/static/src/models/message_view.js

```diff
@@ -13,10 +13,12 @@ registerModel({
          * Briefly highlights the message.
          */
         highlight() {
-            this.update({
-                highlightTimer: [clear(), insertAndReplace()],
-                isHighlighted: true,
-            });
+            this.update({ isHighlighted: true });
+            if (!this.highlightTimer) {
+                this.update({ highlightTimer: {} });
+            } else {
+                this.highlightTimer.update({ doReset: true });
+            }
```

Comment:
Note that I think all this can be written in a single statement:
```js
this.update({
   highlightTimer: { doReset: this.highlightTimer ? true : undefined },
});
```